File: 2_SEMESTRE/codeTS/src/Atividade_pratica_01/Funcionario.ts
import Cargo from './Cargo';
import Login from './Login';
import Pessoa from './Pessoa';

export default class Funcionario extends Pessoa {
  private _cargo: Cargo;
  private _salario: number;
  private _login: Login;
  constructor(
    id: string,
    nome: string,
    email: string,
    telefone: string,
    cpf: string,
    rg: string,
    dataNascimento: string,
    cargo: Cargo,
    salario: number,
    login: Login,
  ) {
    super(id, nome, email, telefone, cpf, rg, dataNascimento);
    (this._cargo = cargo), (this._salario = salario), (this._login = login);
  }

  get cargo(): Cargo {
    return this._cargo;
  }
  set cargo(valor: Cargo) {
    this._cargo = valor;
  }
  get salario(): number {
    return this._salario;
  }
  set salario(valor: number) {
    this._salario = valor;
  }
  get login(): Login {
    return this._login;
  }
  set login(valor: Login) {
    this._login = valor;
  }
}

File: 3_SEMESTRE/codePY/funcionario.py
from pessoa import Pessoa


class Funcionario(Pessoa):
    def __init__(
        self,
        _id: str,
        nome: str,
        cpf: str,
        telefone: str,
        endereco: str,
        funcao: str,
        setor: str,
    ):
        super().__init__(_id, nome, cpf, telefone)
        self.__endereco = endereco
        self.__funcao = funcao
        self.__setor = setor

    @property
    def endereco(self) -> str:
        return self.__endereco

    @endereco.setter
    def endereco(self, endereco: str):
        self.__endereco = endereco

    @property
    def funcao(self) -> str:
        return self.__funcao

    @funcao.setter
    def funcao(self, funcao: str):
        self.__funcao = funcao

    @property
    def setor(self) -> str:
        return self.__setor

    @setor.setter
    def setor(self, setor: str):
        self.__setor = setor
